src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -175,21 +175,29 @@ public void visitToken(DetailAST ast) {
 
         if (rcurly != null) {
             final String violation;
-            if (shouldStartLine) {
-                final String targetSourceLine = getLines()[rcurly.getLineNo() - 1];
-                violation = validate(details, option, true, targetSourceLine);
-            }
-            else {
-                violation = validate(details, option, false, "");
-            }
-
+            violation = validate(details, option,
+                    shouldStartLine, getTargetSourceLine(rcurly.getLineNo() - 1));
             if (!violation.isEmpty()) {
                 log(rcurly, violation, "}", rcurly.getColumnNo() + 1);
             }
         }
     }
 
     /**
+     * Fetches line based on passed line no.
+     * @param lineNo for fetching lines.
+     * @return lines associated with line no.
+     */
+    private String getTargetSourceLine(int lineNo) {
```

Comment:
> what we will pass at the time of function call 
> I'd pass there something like a line number

`getTargetSourceLine` is getting it's line number from `rcurly` which is stored in `details`, which we are already passing to `validate` and almost every method in this check.
It seems to me when can remove `String targetSourceLine` parameter from the `validate` method, remove `String targetSourceLine` parameter from `isOnStartOfLine`, and grab and examine the line directly in `isOnStartOfLine`.
This is the only area using this target line currently. So we don't need to pass it around to everyone. If we were using it more, then I would consider moving it into `details`, but we aren't.
@Vladlis What do you think?